### PR TITLE
Generate vBuilder method

### LINKED ISSUE #362: Generate `vBuilder()` shortcut for all messages
To have a more neat dev experience it will be really helpful to have `vBuilder()` static shortcut method that will create a new instance of an appropriate `VBuilder`, e.g.:
```
UserIdVBuilder builder = UserId.vBuilder();
```
Also, as the `toVBuilder` instance method should be generated to be able to convert an instance to the builder again:
```
UserIdVBuilder builder = userId.toVBuilder();
```

These methods should be generated for every type that has a `VBuilder` and the code generation should be performed using the newly available `generateMethods` functionality:
```
generateMethods {
    applyFactory "io.spine.tools.protoc.VBuilderMethodFactory", messages().all()
}
```
where `messages().all()` is just a shortcut to the `messages().inFiles(suffix: ".proto")`.

### PR DESCRIPTION
**Pull request by yuri-sergiichuk**
This PR closes #362 issue.

The new `VBuilderMethodFactory` generates:
1.  The `vBuilder` shortcut static method that creates an instance of the corresponding to the message `ValidatingBuilder`.
1. The `toVBuilder` instance method that converts an instance to the corresponding `ValidatingBuilder` with the initial state of the message itself.

### CODE REVIEW

**Comment by yuri-sergiichuk** [comment]
@armiol PTAL.

**Comment by yuri-sergiichuk** [comment]
@dmdashenkov PTAL.

File: base/src/main/java/io/spine/tools/protoc/method/vbuilder/VBuilderMethodFactory.java [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+
+/**
+ * A {@link MethodFactory} used by default to create new static helper methods for all messages.
+ *
+ * <p>Creates {@code public static final VBuilder vBuilder()} method.
+ */
+@Immutable
+public class VBuilderMethodFactory implements MethodFactory {
+
+    public VBuilderMethodFactory() {
```
Comment (dmdashenkov) [inline base/src/main/java/io/spine/tools/protoc/method/vbuilder/VBuilderMethodFactory.java]: Please remove the default ctor.

File: base/src/main/java/io/spine/tools/protoc/method/vbuilder/VBuilderMethodFactory.java [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+/**
+ * A {@link MethodFactory} used by default to create new static helper methods for all messages.
+ *
+ * <p>Creates {@code public static final VBuilder vBuilder()} method.
+ */
+@Immutable
+public class VBuilderMethodFactory implements MethodFactory {
```
Comment (dmdashenkov) [inline base/src/main/java/io/spine/tools/protoc/method/vbuilder/VBuilderMethodFactory.java]: If it's immutable, it should be final.

File: base/src/main/java/io/spine/tools/protoc/method/vbuilder/VBuilderMethodFactory.java [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+        return ImmutableList.of(newVBuilderSpec(vBuilder), newToVBuilderSpec(vBuilder));
+    }
+
+    /**
+     * Creates new {@code public static final T vBuilder()} method where {@code <T>} is a
+     * {@link io.spine.validate.ValidatingBuilder ValidatingBuilder}:
+     * <pre>
+     *     {@code
+     *     public static final T vBuilder(){
+     *         return T.newBuilder();
```
Comment (dmdashenkov) [inline base/src/main/java/io/spine/tools/protoc/method/vbuilder/VBuilderMethodFactory.java]: Please add a real-life example instead. `T` is not of much help here.

File: base/src/test/proto/spine/test/tools/method/vbuilder/messages_with_vbuilders.proto [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+
+option (type_url_prefix) = "type.spine.io";
+option java_package = "io.spine.test.tools.method.vbuilder";
+option java_multiple_files = true;
+option java_outer_classname = "WithVBuildersProto";
+
+// This file defines test environment for `VBuilderMethodFactoryTest`.
+
+// An example event message.
+message ExampleEvent {
```
Comment (dmdashenkov) [inline base/src/test/proto/spine/test/tools/method/vbuilder/messages_with_vbuilders.proto]: Please use a domain.

File: base/src/test/proto/spine/test/tools/method/vbuilder/rejections.proto [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+option (type_url_prefix) = "type.spine.io";
+option (SPI_all) = true;
+option java_package = "io.spine.test.tools.method.vbuilder.rejections";
+// Keep all the rejections under the same outer class called `Rejections`.
+
+// This file defines test environment for `VBuilderMethodFactoryTest`.
+
+// A sample rejection to verify that VBuilderMethodFactory does not generate `vBuilder` method for
+// rejections.
+message SampleRejection {
```
Comment (dmdashenkov) [inline base/src/test/proto/spine/test/tools/method/vbuilder/rejections.proto]: Same as above.

**Comment by yuri-sergiichuk** [comment]
@dmdashenkov PTAL again.

**Pull request merged**